packages/navigator/src/browser/navigator-container.ts

```diff
@@ -41,6 +42,9 @@ export function createFileNavigatorContainer(parent: interfaces.Container): Cont
     child.rebind(TreeDecoratorService).toDynamicValue(ctx => ctx.container.get(NavigatorDecoratorService)).inSingletonScope();
     bindContributionProvider(child, NavigatorTreeDecorator);
 
+    child.bind(FileNavigatorSearch).toSelf().inSingletonScope();
+    child.bind(NavigatorTreeDecorator).toService(FileNavigatorSearch);
```

Comment:
[question]: Moving this (from the `navigator-frontend-module.ts`) here ensures that we have a new search and decorator instance each time we create a new widget after a disposal. Is that correct? If no, could you explain? Thanks!

Reply:
`FileNavigatorSearch` injects `Tree` which should exists only in the scope of the navigator container, otherwise even the navigator widget is disposed, a reference to it kept from `FileNavigatorSearch`.